src/EnergyPlus/Pumps.cc

```diff
@@ -2366,59 +2208,49 @@ void ReportPumps(EnergyPlusData &state, int const PumpNum)
     // This subroutine sets the pump reporting variables.
 
     // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
-    int InletNode;     // pump inlet node number
     int OutletNode;    // pump outlet node number
     PumpType PumpType; // Current pump type
 
-    PumpType = state.dataPumps->PumpEquip(PumpNum).pumpType;
-    InletNode = state.dataPumps->PumpEquip(PumpNum).InletNodeNum;
-    OutletNode = state.dataPumps->PumpEquip(PumpNum).OutletNodeNum;
-
-    if (state.dataPumps->PumpMassFlowRate <= DataBranchAirLoopPlant::MassFlowTolerance) {
-        state.dataPumps->PumpEquipReport(PumpNum).PumpMassFlowRate = 0.0;
-        state.dataPumps->PumpEquipReport(PumpNum).PumpHeattoFluid = 0.0;
-        state.dataPumps->PumpEquipReport(PumpNum).OutletTemp = state.dataLoopNodes->Node(OutletNode).Temp;
-        state.dataPumps->PumpEquip(PumpNum).Power = 0.0;
-        state.dataPumps->PumpEquip(PumpNum).Energy = 0.0;
-        state.dataPumps->PumpEquipReport(PumpNum).ShaftPower = 0.0;
-        state.dataPumps->PumpEquipReport(PumpNum).PumpHeattoFluidEnergy = 0.0;
-        state.dataPumps->PumpEquipReport(PumpNum).ZoneTotalGainRate = 0.0;
-        state.dataPumps->PumpEquipReport(PumpNum).ZoneTotalGainEnergy = 0.0;
-        state.dataPumps->PumpEquipReport(PumpNum).ZoneConvGainRate = 0.0;
-        state.dataPumps->PumpEquipReport(PumpNum).ZoneRadGainRate = 0.0;
-        state.dataPumps->PumpEquipReport(PumpNum).NumPumpsOperating = 0;
+    auto &thisPump = state.dataPumps->PumpEquip(PumpNum);
+    auto &thisPumpRep = state.dataPumps->PumpEquipReport(PumpNum);
+
+    PumpType = thisPump.pumpType;
+    OutletNode = thisPump.OutletNodeNum;
+    auto &thisOutNode = state.dataLoopNodes->Node(OutletNode);
+    auto &daPumps = state.dataPumps;
+
+    if (daPumps->PumpMassFlowRate <= DataBranchAirLoopPlant::MassFlowTolerance) {
+        new (&(state.dataPumps->PumpEquipReport(PumpNum))) ReportVars();
+        thisPumpRep.OutletTemp = thisOutNode.Temp;
```

Comment:
I'm confused by this one (as in I'm not sure what that construct does). Could you explain the `new` line please? I'm getting lost.

Reply:
I think this basically resets the variable/structure back to its default value.  In this case, it zeroes everything out.  The code it replace did that with one exception (OutletTemp)., hence the next line.

Reply:
This "new with placement". It's been a C++ feature since the beginning. By giving an address argument to `new`, you suppress the memory allocation and call the constructor (in this case `ReportVars()`) on the address provided. 

It's similar to `state.dataPumps->PumpEquipReport(PumpNum) = ReportVars();` but it doesn't construct and copy, it just constructs into the destination. Yes, I know about RVO. But why rely on RVO when you have this?

Reply:
Today I learned! Thanks